src/lib/io/sol-memmap-storage.c

```diff
@@ -343,19 +361,160 @@ check_map(const struct sol_memmap_map *map)
     return true;
 }
 
+#ifdef USE_I2C
+static char *
+sol_slice_to_str(const struct sol_str_slice *slice)
```

Comment:
add this function as a helper in `sol-str-slice.h`, also fix the name to be `sol_str_slice_to_string()`.